docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,530 @@
+      env:
+        - name: CPU_LIMIT
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.containers[?(@.name=="container-name")].resources.limits.cpu
+```
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kubernetes-downwardapi-volume-example
+spec:
+  containers:
+    - name: client-container
+      image: gcr.io/google_containers/busybox command: ["sh", "-c", "while true; do if [[ -e /etc/labels ]]; then cat /etc/labels; fi; if [[ -e /etc/annotations ]]; then cat /etc/annotations; fi;sleep 5; done"]
+      resources:
+        requests:
+          memory: "64Mi"
+          cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      volumeMounts:
+        - name: podinfo
+          mountPath: /etc
+          readOnly: false
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "cpu_limit"
+            fieldRef:
+              fieldPath: spec.containers[?(@.name=="container-name")].resources.limits.cpu
+```
+
+### API with partial json path selectors
+
+Partial json path selectors specify paths to resources limits and requests
+relative to the container spec.
+
+#### Environment variables
+
+This table shows how selectors can be used for various requests and
+limits to be exposed as environment variables.
+
+| Name | Selectors |
+| -------------------- | -------------------|
+| CPU_LIMIT | resources.limits.cpu |
+| MEMORY_LIMIT |resources.limits.memory |
+| CPU_REQUEST | resources.requests.cpu |
+|MEMORY_REQUEST | resources.requests.memory |
+
+Since environment variables are container scoped, so there is no need
+to specify container name as part of the partial selectors as they are
+relative to container spec.
+
+#### Volume plugin
+
+| Path | Selectors |
+| -------------------- | -------------------|
+| container_name/cpu_limit | resources.limits.cpu |
+|container_name/memory_limit | resources.limits.memory |
+| container_name/cpu_request | resources.requests.cpu |
+|container_name/memory_request | resources.requests.memory |
+
+Since environment variables are container scoped, so container name must
+be specified as part of their path name. The format is that container
+name and path name must be separated by slash (`/`).
+
+Note: Also, environment variables and volume path names are examples
+only and not necessarily as specified above, and the selectors do not
+have to start with dot.
+
+Partial json selectors will be implemented by introducing
+`containerFieldRef` to extend the current implementation for resources
+requests and limits, and will be part of `type DownwardAPIVolumeFile`
+and `type EnvVarSource` as follows:
+
+```
+// Represents a single file containing information from the downward API
+type DownwardAPIVolumeFile struct {
+     // Required: Path is  the relative path name of the file to be created. Must not be absolute or contain the '..' path. Must be utf-8 encoded. The first item of the relative path must not start with '..'
```

Comment:
The comment is confusing: Must not be absolute or contain the '..'

Does that mean it must or can't contain `..` if not absolute? If the latter, please s/or/nor/ and remove the redundant sentence about not starting with `..`.